salt/orch/etcd-migrate.sls

```diff
@@ -0,0 +1,155 @@
+# To migrate etcd storage version for Kubernestes execute following steps on Admin node:
+# 1. docker exec -ti $(docker ps | grep salt-master | cut -d" " -f1) salt-run state.orchestrate orch.etcd-migrate
+# 2. btrfs property set -ts /.snapshots/1/snapshot ro false
+# 3. mount -o remount,rw /
+# 4. sed -i '/etcd_version/ s/etcd2/etcd3/' /usr/share/salt/kubernetes/pillar/params.sls
+# 5. docker exec -ti $(docker ps | grep salt-master | cut -d" " -f1) salt '*' saltutil.refresh_pillar
+# 6. docker exec -ti $(docker ps | grep salt-master | cut -d" " -f1) salt '*' salt.highstate
+
+# Generic Updates
+update_pillar:
+  salt.function:
+    - tgt: '*'
+    - name: saltutil.refresh_pillar
+
+update_grains:
+  salt.function:
+    - tgt: '*'
+    - name: saltutil.refresh_grains
+
+update_mine:
+  salt.function:
+    - tgt: '*'
+    - name: mine.update
+    - require:
+       - salt: update_pillar
+       - salt: update_grains
+
+update_modules:
+  salt.function:
+    - name: saltutil.sync_modules
+    - tgt: '*'
+    - kwarg:
+        refresh: True
+    - require:
+      - salt: update_mine
+
+{%- set masters = salt.saltutil.runner('mine.get', tgt='G@roles:kube-master', fun='network.ip_addrs', tgt_type='compound') %}
+{%- for master_id in masters.keys() %}
+
+{{ master_id }}-stop-services:
+  salt.state:
+    - tgt: {{ master_id }}
+    - sls:
+      - kube-apiserver.stop
+      - kube-controller-manager.stop
+      - kube-scheduler.stop
+      - docker.stop
+      - flannel.stop
+      - etcd.stop
+
+{{ master_id }}-wait-action:
```

Comment:
I'm not sure I understand this (and the other wait-actions) - they don't actually do anything, and we don't even wait for the "sleep 1" to complete before proceeding due to the `bg: True` kwarg?

Reply:
Fixed